hapi/hedera-protobufs/services/state/blockstream/block_stream_info.proto

```diff
@@ -131,4 +132,27 @@ message BlockStreamInfo {
      * the current block.
      */
     proto.Timestamp block_end_time = 9;
+
+    /**
+     * Whether the post-upgrade work has been done.
+     * <p>
+     * This MUST be false if and only if the network just restarted
+     * after an upgrade and has not yet done the post-upgrade work.
+     */
+    bool post_upgrade_work_done = 10;
+
+    /**
+     * A version describing the version of application software.
+    * <p>
+    * This SHALL be the software version that created this block.
+    */
+    proto.SemanticVersion creation_software_version = 11;
```

Comment:
Because we have this , we don't need to depend on `freeze_time` anymore to detect if this is the first transaction we're handling after a freeze restart in `BlockStreamManagerImpl`?

Reply:
Exactly!